simplenote_cli/sncli.py

```diff
@@ -1193,6 +1193,65 @@ def cli_note_markdown(self, key, markdown):
         self.ndb.set_note_markdown(key, markdown)
         self.sync_notes()
 
+    def cli_note_tags_get(self, key):
+
+        note = self.ndb.get_note(key)
+        if not note:
+            self.log('ERROR: Key does not exist')
+            return
+
+        tags = utils.get_note_tags(note)
+        return tags
+
+    def cli_note_tags_set(self, key, tags):
+
+        note = self.ndb.get_note(key)
+        if not note:
+            self.log('Error: Key does not exist')
+            return
+
+        self.ndb.set_note_tags(key, tags)
+        self.sync_notes()
+
+    def cli_note_tags_add(self, key, new_tags):
+
+        note = self.ndb.get_note(key)
+        if not note:
+            self.log('Error: Key does not exist')
+            return
+
+        # Add tag only if it isn't already there
+        old_tags = self.cli_note_tags_get(key)
+        if old_tags:
+            old_tag_list = old_tags.split(',')
+            new_tag_list = new_tags.split(',')
+            tag_list = old_tag_list
+            for tag in new_tag_list:
+                if tag not in tag_list:
+                    tag_list.append(tag)
+            tags = ','.join(tag_list)
+        else:
+            tags = new_tags
+
+        self.cli_note_tags_set(key, tags)
+
+    def cli_note_tags_rm(self, key, rm_tags):
+
+        note = self.ndb.get_note(key)
+        if not note:
+            self.log('Error: Key does not exist')
+            return
+
+        old_tags = self.cli_note_tags_get(key)
+        if old_tags:
+            old_tag_list = old_tags.split(',')
+            rm_tag_list = rm_tags.split(',')
+            tag_list = old_tag_list
+            for tag in rm_tag_list:
+                if tag in tag_list:
```

Comment:
This is case-sensitive matching, while it seems the simplenote api (or some internals of sncli) is case insensitive. Perhaps here and above in adding tags should be case-insensitive matching?

Reply:
Yes, good call. I'll make the change when I'm back at my desk tomorrow.